airbyte-integrations/connectors/source-the-guardian-api/unit_tests/test_paginator.py

```diff
@@ -0,0 +1,72 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import logging
+from unittest.mock import MagicMock
+
+import airbyte_cdk
+import pytest
+import requests
+from airbyte_cdk.test.utils.manifest_only_fixtures import components_module, connector_dir
+
+logger = logging.getLogger(__name__)
+
+def create_response(current_page: int, total_pages: int) -> requests.Response:
+    """Helper function to create mock responses"""
+    response = MagicMock(spec=requests.Response)
+    response.json.return_value = {
+        "response": {
+            "currentPage": current_page,
+            "pages": total_pages
+        }
+    }
+    return response
+
+@pytest.mark.parametrize(
+    "current_page,total_pages,expected_next_page",
+    [
+        (1, 5, 2),      # First page
+        (2, 5, 3),      # Middle page
+        (4, 5, 5),      # Second to last page
+        (5, 5, None),   # Last page
+        (1, 1, None),   # Single page
+    ],
+    ids=["First page", "Middle page", "Penultimate page", "Last page", "Single page"]
+)
+def test_page_increment(connector_dir, components_module, current_page, total_pages, expected_next_page):
```

Comment:
@ChristoGrab I see something sus.

Since you import `components_module` at the top and reference it as an argument.

Which is it using?

What happens if you remove the argument version?
```suggestion
def test_page_increment(connector_dir, current_page, total_pages, expected_next_page):
```

Reply:
Switched to using the plugin approach and it works without the import statement 🎉